### ISSUE TITLE
BUG: update the internal syscall tables for Linux v5.1

### ISSUE THREAD

**Issue by pcmoore** [seccomp/libseccomp#136]
This is pretty self-explanatory, but I'm creating an issue so we don't forget to update the internal syscall tables before the next libseccomp release.

**Comment by drakenclimber** [comment]
I'm not familiar with the backstory of adding s390 syscalls into other architectures.

Are they still needed?  The [s390 unistd.h](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/s390/include/uapi/asm/unistd.h?h=v5.0-rc5) no longer has its own list of `__NR_{syscall}`s.

**Comment by drakenclimber** [comment]
They were removed in this commit:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/arch/s390/include/uapi/asm/unistd.h?h=v5.0-rc5&id=4381f9f12e79436f90b14f326817cb1ed978940d

**Comment by pcmoore** [comment]
It looks like s390[x] has changed over to using s390/kernel/syscalls/syscall.tbl, we'll need to update the src/arch-syscall-validate script, but we can probably just borrow/steal the ARM code in that file.

Looking at the following we may have to update our ppc and parisc code too.

```
# find . -name "syscall.tbl"
./arm/tools/syscall.tbl
./sh/kernel/syscalls/syscall.tbl
./s390/kernel/syscalls/syscall.tbl
./m68k/kernel/syscalls/syscall.tbl
./ia64/kernel/syscalls/syscall.tbl
./microblaze/kernel/syscalls/syscall.tbl
./xtensa/kernel/syscalls/syscall.tbl
./powerpc/kernel/syscalls/syscall.tbl
./parisc/kernel/syscalls/syscall.tbl
./alpha/kernel/syscalls/syscall.tbl
./sparc/kernel/syscalls/syscall.tbl
```

**Comment by drakenclimber** [comment]
I have appeased `arch-syscall-validate`, but I have one open question that I haven't sufficiently resolved to make me fully comfortable with the changes.

s390 has an ABI option called `spu` in [their table](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/powerpc/kernel/syscalls/syscall.tbl?h=v5.0-rc5#n34) and I'm not sure how to resolve it.  There are syscalls like `umount` where there is a 32-bit, a 64-bit, and a spu entry.  Does this mean that it is a separate architecture and should have its own entry alongside ppc32 and ppc64?

**Comment by pcmoore** [comment]
Wow, s390 just keeps getting weirder and weirder.

Unfortunately, my googling around for s390 and "spu" doesn't arrive at anything definitive.  My *guess* is that we care about "common" and "nospu" for both s390 and s390x, adding in "32" and "64" for s390 and s390x respectively.

It is probably a good idea to ping the s390 folks to get a definitive answer (below taken from the kernel's MAINTAINERS file):

    S390
    M:      Martin Schwidefsky <schwidefsky@de.ibm.com>
    M:      Heiko Carstens <heiko.carstens@de.ibm.com>
    L:      linux-s390@vger.kernel.org
    W:      http://www.ibm.com/developerworks/linux/linux390/

**Comment by drakenclimber** [comment]
Will do.  Thanks!

**Comment by drakenclimber** [comment]
Bah - I was multitasking too much on Friday.  Sorry about that.  :(

PPC introduced the "spu" moniker - not 390.  Argh.

**Comment by pcmoore** [comment]
For posterity's sake, here is the upstream ppc mailing list discussion:

* https://lists.ozlabs.org/pipermail/linuxppc-dev/2019-February/185336.html

... it appears that we can safely ignore "spu" right now.

**Comment by pcmoore** [comment]
@drakenclimber I see a patch in the log above - is this ready for a PR?

**Comment by pcmoore** [comment]
Resolved via bd42d36c9b9f4e892a1d30c192dcbd11a5b7f1dd.
